### PR TITLE
fix: Implemented Rate Limiter against Message Flooding.

### LINKED ISSUE #470: Implement Rate Limiting for Messages.
### Description

Without rate limiting, users can flood the chat with messages, potentially leading to performance issues and misuse. Rate limiting will ensure that users can send messages at a reasonable pace and prevent abuse.

### Screenshots

anon_chat

### Additional information

I propose implementing rate limiting for message sending. A rate limit of, for example, 70 messages per minute per user, can be a good starting point .

### 🥦 Browser

Google Chrome

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by Tholkappiar**
# Fixes Issue

**My PR closes #470**

# 👨‍💻 Changes proposed(What did you do ?)
Implemented Rate Limiter against Message Flooding.
# ✔️ Check List (Check all the applicable boxes)

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

When the user send 25 messages within a minute this will limit or block the messages - for message flooding and spamming 

# 📷 Screenshots

Screenshot (87)

### CODE REVIEW

File: server/sockets/sendMessage.js [hunk 1]
```diff
@@ -26,19 +22,27 @@ module.exports = (socket) => {
...
+          message:
+            'You have exceeded the message limit. Please try again later.',
+        });
         return;
       }
 
-      /**
-       * Cache the sent message in memory a nd persist to db
-       */
+      // Cache the sent message in memory and persist to the database
```
Comment (Dun-sin) [inline server/sockets/sendMessage.js:40]: don't change this

File: server/sockets/sendMessage.js [hunk 2]
```diff
@@ -26,19 +22,27 @@ module.exports = (socket) => {
...
       const sentMessage = await addMessage(chatId, {
         message,
         time,
         senderId,
         type: 'message',
-        containsBadword
```
Comment (Dun-sin) [inline server/sockets/sendMessage.js:41]: you can't just remove an important code when merging a conflict check what both code does and don't just delete anything

File: server/sockets/sendMessage.js [hunk 3]
```diff
@@ -5,17 +5,13 @@ const {
...
   socket.on(
     NEW_EVENT_SEND_MESSAGE,
-    async ({ senderId, message, time, chatId, containsBadword }, returnMessageToSender) => {
-      // Below line is just a failed message simulator for testing purposes.
-
-      // const rndInt = Math.floor(Math.random() * 6) + 1;
-      // if (rndInt % 2 !== 0) {
-      //   return;
-      // }
-
```
Comment (Dun-sin) [inline server/sockets/sendMessage.js:18]: not part of your issue, pls revert back

**Review by Dun-sin — CHANGES_REQUESTED** [review]
you need to also prevent the user from sending a text in the frontend until the limit resets

**Review by Tholkappiar — COMMENTED** [review]
Just updated the changes , reverted back the previous code which do not fall on my scope , don't know why some changes are only visible in the remote ! Please  kindly review the changes !

**Comment by Dun-sin** [comment]
> you need to also prevent the user from sending a text in the frontend until the limit resets

@Tholkappiar

**Review by Tholkappiar — COMMENTED** [review]
Hey Hello , just implemented the limiter in the frontend part , only changed my issue code , dont know why git says i updated all , kindly check the work which nearly got 3 hrs to update , kindly give the feedback .If all good kindly add up the hacktoberfest things .Thank you !

**Review by Tholkappiar — COMMENTED** [review]
Hey Hello , just implemented the limiter in the frontend part , only changed my issue code , dont know why git says i updated all , kindly check the work which nearly got 3 hrs to update , kindly give the feedback .If all good kindly add up the hacktoberfest things .Thank you !

**Review by Dun-sin — APPROVED** [review]
You did great, thanks for contributing, I hope you will stick around and continue to contribute to this project.

Consider giving this project a star and joining the community discord server if you haven't for more resources and opportunities to connect with others. [👉🏽here👈🏽](https://discord.com/invite/ufcysW9q23)

**Comment by Tholkappiar** [comment]
_Could you pls add the hactoberfest accepted labels for me !_

**Pull request merged**